### PR TITLE
feat: REPUX-703 Add WebSocket server

### PR DESCRIPTION
**Pull request by gbar-repux**

### CODE REVIEW

File: lib/eth-listener.ts [hunk 1]
```diff
@@ -49,8 +60,21 @@ const esClient = require('./elasticsearch/client');
...
+                esClient.update(
+                    {
+                        index: config.elasticsearch.indexes.dataProductEvent,
+                        type: 'data_product_event',
+                        id: event.res.transactionHash,
+                        body: {
+                            doc: event.res,
+                            doc_as_upsert: true,
+                        },
+                        refresh: "wait_for",
```
Comment (kduralek) [inline lib/eth-listener.ts]: single quotes

File: lib/services/data-product-updater.ts [hunk 2]
```diff
@@ -51,9 +51,10 @@ export class DataProductUpdater {
                 id: product.address,
                 body: {
                     doc: product,
-                    doc_as_upsert: true
+                    doc_as_upsert: true,
                 },
-            }
+                refresh: "wait_for",
```
Comment (kduralek) [inline lib/services/data-product-updater.ts]: single quotes

File: lib/services/data-product-updater.ts [hunk 3]
```diff
@@ -65,7 +66,8 @@ export class DataProductUpdater {
                 index: this.esIndexName,
                 type: 'data_product',
                 id: address,
-            }
+                refresh: "wait_for",
```
Comment (kduralek) [inline lib/services/data-product-updater.ts]: single quotes

File: config/config.js.dist [hunk 4]
```diff
@@ -24,6 +27,16 @@ const config = {
         httpUrl: process.env.METAINDEXER_IPFS_HTTP_URL || 'http://repux-indexer-ipfs:8080',
         maxMetaFileSize: process.env.METAINDEXER_IPFS_MAX_META_FILE_SIZE || 512000
     },
+    socketio: {
```
Comment (przemczan) [inline config/config.js.dist]: I would rather named it like `events_socket` or something more descriptive to the functionality it provides. `socket.io` is the name of the library you used, so what if we would like to add another socket.io based feature? :)

**Pull request merged**